### ISSUE TITLE
iOS client engine creates a new NSURLSession for each request, which can result in failures on real devices

### ISSUE THREAD

**Issue by 6bangs** [ktorio/ktor#1341]
**Ktor Version and Engine Used (client or server and name)**
1.2.4, iOS

**Describe the bug**
There seems to be an internal limit on the total number of NSURLSessions allowed per app; it may differ depending on model or OS version, but the limit is definitely around 650 on an iPhone 8 running iOS 11. The following logs appear when the limit is exceeded, and the request fails:

```
nw_path_evaluator_request_nexus_and_stats NECP_CLIENT_ACTION_REQUEST_NEXUS_INSTANCE [28: No space left on device]

nw_endpoint_flow_setup_channel [520.1 2600:9000:2145:4400:7:307d:92c0:21.443 in_progress channel-flow (satisfied)] failed to request nexus instance: Error Domain=kNWErrorDomainPOSIX Code=28 "No space left on device" UserInfo={NSDescription=No space left on device}
```

We have also reproduced the bug on an iPhone X. It does _not_ reproduce in the simulator.

**To Reproduce**
1. Make 650+ requests on a real device
2. Observe the subsequent errors

**Expected behavior**
Requests should continue to succeed as usual.

**Comment by e5l** [comment]
Fixed in `1.3.1`

**Comment by Gargantua7** [comment]
@e5l @6bangs 

I also had this problem in ktor 2.3.12 with `Drawin` engine

Each Request and Response consumes a lot of time and data traffic in TSL Handshake

And sometimes will throws "No space left on device"
```
io.ktor.client.engine.darwin.DarwinHttpRequestException: Exception in http request: Error Domain=NSPOSIXErrorDomain Code=28 "No space left on device" UserInfo={_NSURLErrorFailingURLSessionTaskErrorKey=LocalDataTask <0C3C2E4B-8BDC-451A-B4BD-E2852EBD6006>.<1>, _kCFStreamErrorDomainKey=1, _kCFStreamErrorCodeKey=28, _NSURLErrorRelatedURLSessionTaskErrorKey=(
    "LocalDataTask <0C3C2E4B-8BDC-451A-B4BD-E2852EBD6006>.<1>",
    "LocalDataTask <F62D076D-B593-4608-8E8C-9497C4D703F4>.<1>"
), _NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0, ipv4, dns, proxy}
    at 0   Shared                              0x10610888f        kfun:io.ktor.client.engine.darwin.DarwinHttpRequestException#<init>(platform.Foundation.NSError){} + 231
    at 1   Shared                              0x106613c9b        _696f2e6b746f723a6b746f722d636c69656e742d64617277696e2f6f70742f6275696c644167656e742f776f726b2f386435343762393734613762653231662f6b746f722d636c69656e742f6b746f722d636c69656e742d64617277696e2f64617277696e2f7372632
```
